xdsl/dialects/riscv.py

```diff
@@ -1829,6 +1832,61 @@ def assembly_line(self) -> str | None:
         return _assembly_line(self.directive.data, (value,), is_indented=False)
 
 
+@irdl_op_definition
+class CustomAssemblyInstructionOp(IRDLOperation, RISCVInstruction):
+    """
+    An instruction with unspecified semantics, that can be printed during assembly
+    emission. Can be useful when prototyping or leveraging dynamic features in RISC-V
```

Comment:
Overall this docstring looks great, but could you be a bit more explicit about what you mean with dynamic features in RISC-V emulation? Like e.g. let the emulator prints something directly in emulation?
Or is that not what you mean by this? I'm really sorry, apparently it's still not very clear to me what you mean by this :sweat_smile:

Reply:
You can add custom functions to riscemu that match on strings. Not sure how else to put this without mentioning riscemu directly...

Reply:
Sure, I'll remove it